src/test/java/org/jenkinsci/plugins/workflow/job/WorkflowRunTest.java

```diff
@@ -406,30 +413,58 @@ public void culprits() throws Exception {
         SemaphoreStep.success("waitFirst/1", null);
 
         SemaphoreStep.waitForStart("waitSecond/1", b1);
-        assertEquals(ImmutableSet.of("alice1"), b1.getCulpritIds());
+        assertCulprits(b1, "alice1");
         SemaphoreStep.success("waitSecond/1", null);
 
         SemaphoreStep.waitForStart("waitThird/1", b1);
-        assertEquals(ImmutableSet.of("alice1", "bob1"), b1.getCulpritIds());
+        assertCulprits(b1, "alice1", "bob1");
         SemaphoreStep.failure("waitThird/1", new AbortException());
 
         r.assertBuildStatus(Result.FAILURE, r.waitForCompletion(b1));
 
         WorkflowRun b2 = p.scheduleBuild2(0).waitForStart();
 
         SemaphoreStep.waitForStart("waitFirst/2", b2);
-        assertEquals(ImmutableSet.of("alice1", "bob1"), b2.getCulpritIds());
+        assertCulprits(b2, "alice1", "bob1");
         SemaphoreStep.success("waitFirst/2", null);
 
         SemaphoreStep.waitForStart("waitSecond/2", b2);
-        assertEquals(ImmutableSet.of("alice1", "bob1", "alice2"), b2.getCulpritIds());
+        assertCulprits(b2, "alice1", "bob1", "alice2");
         SemaphoreStep.success("waitSecond/2", null);
 
         SemaphoreStep.waitForStart("waitThird/2", b2);
-        assertEquals(ImmutableSet.of("alice1", "bob1", "alice2", "bob2"), b2.getCulpritIds());
+        assertCulprits(b2, "alice1", "bob1", "alice2", "bob2");
         SemaphoreStep.success("waitThird/2", b2);
 
         r.assertBuildStatusSuccess(r.waitForCompletion(b2));
-        assertEquals(ImmutableSet.of("alice1", "bob1", "alice2", "bob2", "charlie2"), b2.getCulpritIds());
+        assertCulprits(b2, "alice1", "bob1", "alice2", "bob2", "charlie2");
     }
+
+    private void assertCulprits(WorkflowRun b, String... expectedIds) throws IOException, SAXException {
+        Set<String> actual = new TreeSet<>();
+        for (String u : b.getCulpritIds()) {
+            actual.add(u);
+        }
+        assertEquals(actual, new TreeSet<>(Arrays.asList(expectedIds)));
+
+        if (expectedIds.length > 0) {
+            JenkinsRule.WebClient wc = r.createWebClient();
+            WebResponse response = wc.goTo(b.getUrl() + "api/json?tree=culprits[id]", "application/json").getWebResponse();
+            JSONObject json = JSONObject.fromObject(response.getContentAsString());
+
+            Object culpritsArray = json.get("culprits");
+            assertNotNull(culpritsArray);
```

Comment:
From here to the end of the method could probably be simplified to one `assertEquals`, after preparing the expected `JSONArray`.